src/ai/backend/common/events.py

```diff
@@ -1429,24 +1443,46 @@ async def _subscribe_loop(self) -> None:
 class EventProducer:
     _closed: bool
     _msg_queue: AbstractMessageQueue
+    _broadcaster: AbstractBroadcaster
     _log_events: bool
 
     def __init__(
         self,
         msg_queue: AbstractMessageQueue,
+        broadcaster: AbstractBroadcaster,
         *,
         source: AgentId,
         log_events: bool = False,
     ) -> None:
         self._closed = False
         self._msg_queue = msg_queue
+        self._broadcaster = broadcaster
         self._source_bytes = source.encode()
         self._log_events = log_events
 
     async def close(self) -> None:
         self._closed = True
         await self._msg_queue.close()
 
+    async def broadcast_event(
+        self,
+        event: AbstractEvent,
+        source_override: Optional[AgentId] = None,
+    ) -> None:
+        if self._closed:
+            return
+        source_bytes = self._source_bytes
+        if source_override is not None:
+            source_bytes = source_override.encode()
+
+        raw_event = {
+            b"name": event.name.encode(),
+            b"source": source_bytes,
+            b"args": msgpack.packb(event.serialize()),
+        }
+        b = dump_json(raw_event)
+        await self._broadcaster.broadcast(b)
```

Comment:
```suggestion
        await self._broadcaster.broadcast(dump_json(raw_event))
```

Reply:
It is intended that only one action is performed on a line.
When more than one action is performed, it's hard to see when an issue occurs.